### PR TITLE
fix,howto/addition-of-a-REST-API-Walkthrough

### PR DESCRIPTION
**Pull request by jjargot**
Addition of a walkthrough example with X-Bonita-API-Token

Versions: 7.4, 7.5

### CODE REVIEW

File: md/multi-tenancy-and-tenant-configuration.md [hunk 1]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+
+#### Walk-through
+##### Install `curl` command line tool
+`curl` is available on Linux OS and it transfers data from or to a server with various protocols such as HTTP and HTTPS, see manual page.
+
+    $ sudo apt install curl
+
+NOTE: this is to be done only once.
+##### Start a tomcat
+Download a BonitaBPMSubscription-7.*-Tomcat-7.0.76.zip
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: escape the * (markdown special character):
\\\*

File: md/multi-tenancy-and-tenant-configuration.md [hunk 2]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
 
 After the first login, default theme will be loaded and the login page will look as usual. 
 :::
+### REST API
+
+The [platform REST API](platform-api.md) is a REST layer around the Java PlatformAPI to create the tenant.
+
+#### Walk-through
+##### Install `curl` command line tool
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: a line break before this title would improve readability

File: md/multi-tenancy-and-tenant-configuration.md [hunk 3]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+`curl` is available on Linux OS and it transfers data from or to a server with various protocols such as HTTP and HTTPS, see manual page.
+
+    $ sudo apt install curl
+
+NOTE: this is to be done only once.
+##### Start a tomcat
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: a line break before this title would improve readability

File: md/multi-tenancy-and-tenant-configuration.md [hunk 4]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
 
 After the first login, default theme will be loaded and the login page will look as usual. 
 :::
+### REST API
+
+The [platform REST API](platform-api.md) is a REST layer around the Java PlatformAPI to create the tenant.
+
+#### Walk-through
+##### Install `curl` command line tool
+`curl` is available on Linux OS and it transfers data from or to a server with various protocols such as HTTP and HTTPS, see manual page.
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: I would remove "see manual page"

File: md/multi-tenancy-and-tenant-configuration.md [hunk 5]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+    $ sudo apt install curl
+
+NOTE: this is to be done only once.
+##### Start a tomcat
+Download a BonitaBPMSubscription-7.*-Tomcat-7.0.76.zip
+
+Unzip it and provide a valid license file
+
+Start the tomcat
+##### Login
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: a line break before this title would improve readability

File: md/multi-tenancy-and-tenant-configuration.md [hunk 6]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+
+    $ sudo apt install curl
+
+NOTE: this is to be done only once.
+##### Start a tomcat
+Download a BonitaBPMSubscription-7.*-Tomcat-7.0.76.zip
+
+Unzip it and provide a valid license file
+
+Start the tomcat
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: I would use bullet points for the 3 previous points

File: md/multi-tenancy-and-tenant-configuration.md [hunk 7]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+    < Set-Cookie: JSESSIONID=46EF8A05819B6C268EE700F3C3FC939A; Path=/bonita/; HttpOnly
+
+    < Set-Cookie: X-Bonita-API-Token=a94cbf84-6b71-409a-981f-f91b17466929; Path=/
+
+The response to this REST API call (HTTP) generates 2 cookies, which must be transfered with each subsequent calls.
+One of the cookie is `X-Bonita-API-Token`.
+
+Note that the security against CSRF attacks is enabled by default for all fresh installations; the subsequence REST API calls using DELETE, POST, or PUT HTTP methods must define the `X-Bonita-API-Token` header, with the value transmitted via the associated cookie.
 
+The cookies had been saved on the disk, in the `saved_cookies.txt` file:
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: "have" instead of "had"

File: md/multi-tenancy-and-tenant-configuration.md [hunk 8]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+
+    $ cat saved_cookies.txt 
+    
+    #HttpOnly_localhost	FALSE	/bonita/	FALSE	0	JSESSIONID	46EF8A05819B6C268EE700F3C3FC939A
+    localhost	FALSE	/	FALSE	0	X-Bonita-API-Token	a94cbf84-6b71-409a-981f-f91b17466929
+##### Create the new tenaant
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: typo : tenant (remove one 'a')
a line break before this title would improve readability

File: md/multi-tenancy-and-tenant-configuration.md [hunk 9]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+      "creation": "2017-06-09 15:11:01.191",
+      "icon": "/default.png",
+      "username": "",
+      "description": "My tenant",
+      "name": "MyTenant",
+      "state": "DEACTIVATED",
+      "password": ""
+    }
+The new tenant has the id `101` and its state is `DEACTIVATED`
+##### Activate the tenant with id `101`
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: a line break before this title would improve readability

File: md/multi-tenancy-and-tenant-configuration.md [hunk 10]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+      "password": "",
+      "name": "MyTenant",
+      "icon": "/default.png",
+      "description": "My tenant",
+      "id": "101",
+      "state": "ACTIVATED",
+      "creation": "2017-06-09 15:11:01.191",
+      "username": ""
+    }
+##### Logout
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: a line break before this title would improve readability

File: md/multi-tenancy-and-tenant-configuration.md [hunk 11]
```diff
@@ -55,8 +55,95 @@ Therefore, as long as you have not logged in to the new tenant, the login page w
...
+
+    $ curl -v -b saved_cookies.txt -X GET --url 'http://localhost:8080/bonita/platformlogoutservice?redirect=false'
 ### Java PlatformAPI
+This solution can be used when the portal is not needed: it creates the tenant at engine side only.
```
Comment (abirembaut) [inline md/multi-tenancy-and-tenant-configuration.md]: I think we should either remove the "at" or say "at engine level only" instead of "side"

**Review by abirembaut — CHANGES_REQUESTED** [review]
Great !
It looks good to me. I just have a few comments

File: md/multi-tenancy-and-tenant-configuration.md (hunk 2 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 4 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 3 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 1 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 6 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 5 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 7 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 8 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 9 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 10 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

File: md/multi-tenancy-and-tenant-configuration.md (hunk 11 above)
Comment (jjargot) [reply md/multi-tenancy-and-tenant-configuration.md]: done

**Review by abirembaut — APPROVED** [review]

**Comment by abirembaut** [comment]
:+1:

**Pull request merged**